From dc52d84e947b5c050236c6dc7929d457a8390d3b Mon Sep 17 00:00:00 2001
From: spycat88 <spycat88@users.noreply.github.com>
Date: Wed, 24 Jan 2024 11:37:48 +0000
Subject: [PATCH 04/38] arm64: meson: odroid-go-ultra: add over-clocking
 frequencies, specify as turbo mode so they are disabled by default and
 increase vddcpu_a voltage regulator as needed

---
 .../amlogic/meson-g12b-odroid-go-ultra.dts    | 31 +++++++++++++++++--
 1 file changed, 29 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts
index 571a0393e148..c3af3954a1e7 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-go-ultra.dts
@@ -347,7 +347,12 @@ opp-1896000000 {
 		opp-microvolt = <987500>;
 	};
 	opp-1992000000 {
-		opp-microvolt = <1012500>;
+		opp-microvolt = <1050000>;
+	};
+	opp-2016000000 {
+		opp-hz = /bits/ 64 <2016000000>;
+		opp-microvolt = <1050000>;
+		turbo-mode;
 	};
 };
 
@@ -377,6 +382,28 @@ opp-1800000000 {
 	opp-1908000000 {
 		opp-microvolt = <1025000>;
 	};
+	opp-2016000000 {
+		opp-hz = /bits/ 64 <2016000000>;
+		opp-microvolt = <1025000>;
+	};
+	opp-2100000000 {
+		opp-hz = /bits/ 64 <2100000000>;
+		opp-microvolt = <1025000>;
+	};
+	opp-2208000000 {
+		opp-hz = /bits/ 64 <2208000000>;
+		opp-microvolt = <1050000>;
+	};
+	opp-2304000000 {
+		opp-hz = /bits/ 64 <2304000000>;
+		opp-microvolt = <1050000>;
+		turbo-mode;
+	};
+	opp-2400000000 {
+		opp-hz = /bits/ 64 <2400000000>;
+		opp-microvolt = <1050000>;
+		turbo-mode;
+	};
 };
 
 &i2c_AO {
@@ -407,7 +434,7 @@ vddcpu_a: DCDC_REG1 {
 				regulator-always-on;
 				regulator-boot-on;
 				regulator-min-microvolt = <775000>;
-				regulator-max-microvolt = <1025000>;
+				regulator-max-microvolt = <1050000>;
 				regulator-ramp-delay = <6001>;
 				regulator-state-mem {
 					regulator-on-in-suspend;
-- 
2.34.1

